refactor(localize): remove deprecated `canParse` method from TranslationParsers

alan-agius4 opened the pull request:
This change removed the deprecated `canParse` method from all the TranslationParsers.

BREAKING CHANGE: 

- `canParse` method has been removed from all translation parsers in `@angular/localize/tools`. `analyze` should be used instead. 
-  the `hint` parameter in the `parse` methods is now mandatory.

jessicajaniuk reviewed (approved):
reviewed-for: public-api

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

alan-agius4 commented on packages/localize/tools/test/translate/translation_files/translation_parsers/xliff1_translation_parser_spec.ts:
```diff
@@ -6,793 +6,782 @@
+        `    </body>`,
+        `  </file>`,
+        `</xliff>`,
+      ].join('\n');
```
Totally agree.

alan-agius4 committed:
refactor(localize): remove deprecated `canParse` method from TranslationParsers

This change removed the deprecated `canParse` method from all the TranslationParsers.

BREAKING CHANGE:

- `canParse` method has been removed from all translation parsers in `@angular/localize/tools`. `analyze` should be used instead.
-  the `hint` parameter in the`parse` methods is now mandatory.

AndrewKushnir commented:
This PR was merged into the repository by commit d36fd3d9e41984818af5e1aad5e3004574d837bb.
